From 89d35ab9cc29c74a07f93f7f2bad09c07b536c61 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Mon, 23 Aug 2021 16:03:59 +0800
Subject: [PATCH] eulerfs: add Kconfig and Makefile
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 89d35ab9cc29c74a07f93f7f2bad09c07b536c61


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I40JRR
CVE: NA

--------------------------------------

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
Reviewed-by: Hou Tao <houtao1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/Kconfig          |  1 +
 fs/Makefile         |  1 +
 fs/eulerfs/Kconfig  | 10 ++++++++++
 fs/eulerfs/Makefile |  9 +++++++++
 4 files changed, 21 insertions(+)
 create mode 100644 fs/eulerfs/Kconfig
 create mode 100644 fs/eulerfs/Makefile

diff --git a/fs/Kconfig b/fs/Kconfig
index 83c56571d0bc..3cc647e00f3c 100644
--- a/fs/Kconfig
+++ b/fs/Kconfig
@@ -354,6 +354,7 @@ source "fs/cifs/Kconfig"
 source "fs/coda/Kconfig"
 source "fs/afs/Kconfig"
 source "fs/9p/Kconfig"
+source "fs/eulerfs/Kconfig"
 
 endif # NETWORK_FILESYSTEMS
 
diff --git a/fs/Makefile b/fs/Makefile
index 01d36815c3a5..fec76c1b4e06 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -139,3 +139,4 @@ obj-$(CONFIG_RESCTRL)		+= resctrlfs.o
 obj-$(CONFIG_EROFS_FS)		+= erofs/
 obj-$(CONFIG_VBOXSF_FS)		+= vboxsf/
 obj-$(CONFIG_ZONEFS_FS)		+= zonefs/
+obj-$(CONFIG_EULER_FS)		+= eulerfs/
diff --git a/fs/eulerfs/Kconfig b/fs/eulerfs/Kconfig
new file mode 100644
index 000000000000..6328684f0b6a
--- /dev/null
+++ b/fs/eulerfs/Kconfig
@@ -0,0 +1,10 @@
+config EULER_FS
+	tristate "EulerFS: filesystem for non-volatile memories"
+	depends on ARM64 || X86_64
+	depends on FS_DAX
+	select CRC32
+	select LIBCRC32C
+	help
+		EulerFS is NVDIMM filesystem. It uses soft updates and
+		pointer-based dual views to delay synchronous cache flushes
+		and reduce latency significantly in critical path.
diff --git a/fs/eulerfs/Makefile b/fs/eulerfs/Makefile
new file mode 100644
index 000000000000..706e6ebff77e
--- /dev/null
+++ b/fs/eulerfs/Makefile
@@ -0,0 +1,9 @@
+#
+# Makefile for eulerfs
+#
+
+obj-$(CONFIG_EULER_FS) += eulerfs.o
+
+eulerfs-y := dir.o file.o inode.o namei.o super.o symlink.o
+eulerfs-y += dax.o dht.o dep.o nvalloc.o wear.o
+eulerfs-y += kmem_cache.o
-- 
2.26.2

